src/include/plan_node/csv_scan_plan_node.h

```diff
@@ -0,0 +1,181 @@
+#pragma once
+
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+#include "plan_node/abstract_scan_plan_node.h"
+
+namespace terrier::plan_node {
+/**
+ * Plan node for a CSV scan
+ */
+class CSVScanPlanNode : public AbstractScanPlanNode {
```

Comment:
@pmenon Should we rename this `CopyPlanNode`?

Reply:
This node normally has an `Insert` parent node to perform an import. While we can probably collapse it into a single `CopyPlanNode`, I think it's better to leave it as is since it's more descriptive, we can later support scanning files of different formats, and also scan files without the intention of importing the data into the database.